jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/URIUtil.java

```diff
@@ -1582,32 +1428,29 @@ public static void appendSchemeHostPort(StringBuilder url, String scheme, String
      */
     public static void appendSchemeHostPort(StringBuffer url, String scheme, String server, int port)
     {
-        synchronized (url)
-        {
-            url.append(scheme).append("://").append(HostPort.normalizeHost(server));
+        url.append(scheme).append("://").append(HostPort.normalizeHost(server));
 
-            if (port > 0)
+        if (port > 0)
+        {
+            switch (scheme)
             {
-                switch (scheme)
-                {
-                    case "http":
-                        if (port != 80)
-                            url.append(':').append(port);
-                        break;
-
-                    case "https":
-                        if (port != 443)
-                            url.append(':').append(port);
-                        break;
+                case "http":
+                    if (port != 80)
+                        url.append(':').append(port);
+                    break;
 
-                    default:
+                case "https":
+                    if (port != 443)
                         url.append(':').append(port);
-                }
+                    break;
+
+                default:
+                    url.append(':').append(port);
             }
         }
     }
 
-    // Only URIUtil is using this method
+    // Only used by URIUtil
```

Comment:
Make it private then.

Reply:
Can't make it private, as it's tested in URIUtilTest.
And protected isn't valid.

Reply:
Then the comment is wrong, should be `Only used by URIUtil*Test*.`